aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -1042,12 +1042,21 @@ Tensor reduce_sparse_csr_dim0_cpu_template(const Tensor& sparse, ReductionOp rop
   new_crow_indices[1] = nnz;
 
   Tensor new_values = at::empty({nnz}, values.options());
+  bool need_acc = (values.scalar_type() == kHalf || values.scalar_type() == kBFloat16 || values.scalar_type() == kComplexHalf);
+  auto values_acc_option = values.options();
+  if (need_acc) {
+    values_acc_option = values.options().dtype(ScalarType::Float);
```

Comment:
Using `Float` would be wrong when `values.scalar_type() == kComplexHalf` is `true`.